pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -188,7 +188,9 @@ func (d *defaultDriver) Reconcile(ctx context.Context) *reconciler.Results {
 	// always update the elasticsearch state bits
 	d.ReconcileState.UpdateElasticsearchState(*resourcesState, observedState())
 
-	if err := d.verifySupportsExistingPods(resourcesState.CurrentPods); err != nil {
+	_, permissive := d.ES.Annotations[esv1.PermissiveVersionValidation]
+
+	if err := d.verifySupportsExistingPods(resourcesState.CurrentPods); !permissive && err != nil {
```

Comment:
It feels a bit odd to me to have a validation function outside of the `validation` package. I think I would expect them to be placed in the same one. It would allow to have a consistent code path where `verifySupportsExistingPods` would be part of a common validation logic together with `noDowngrades` _(maybe with `verifySupportsExistingPods` disabled in the context of the validation webhook if we don't want to use the cached client)._

That being said I understand that it is not the purpose of this PR and I think I'm fine with the current proposal.

Reply:
I agree. I tried to minimise the changes in this PR given that we want to back port it. How about I open an issue and we fix this in main?